documentation/src/test/java/org/hibernate/validator/referenceguide/chapter09/CustomScriptEvaluatorFactoryForSpringELTest.java

```diff
@@ -0,0 +1,42 @@
+package org.hibernate.validator.referenceguide.chapter09;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import javax.validation.Validation;
+import javax.validation.Validator;
+
+import org.hibernate.validator.HibernateValidator;
+import org.hibernate.validator.constraints.ScriptAssert;
+
+import org.junit.Test;
+
+public class CustomScriptEvaluatorFactoryForSpringELTest {
+
+	@Test
+	public void testCreateNewScriptEvaluator() throws Exception {
+		Validator validator = Validation.byProvider( HibernateValidator.class )
+				.configure()
+				.scriptEvaluatorFactory( new SpringELScriptEvaluatorFactory() )
+				.buildValidatorFactory()
+				.getValidator();
+		assertTrue( validator.validate( new Foo( 1 ) ).isEmpty() );
+		assertEquals( 1, validator.validate( new Foo( -1 ) ).size() );
+	}
+
+	//tag::include[]
+	@ScriptAssert(script = "value > 0", lang = "spring")
```

Comment:
Very nice to see this example for using SpEL! It doesn't seem to work, though, to use the "_this" alias (see `@ScriptAssert#alias()`). Unlike JSR 223, SpEL resolves the properties against a root object, I'm wondering whether we could resolve references against `_this` (or whatever is chosen as name), too, against that root object. It's not mandatory for sure, but it'd be nice to show it for the sake of completeness.

Reply:
We decided against this one as with the Spring syntax, we would end up with `#_this`. I modified the Spring ScriptEvaluatorFactory to inject all the bindings anyway as it seems like a good thing to do.